### PR TITLE
refactor: Use generateSpec to generate fig settings subcommand

### PR DESCRIPTION
**Pull request by QuiiBz**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Refactor

**What is the current behavior? (You can also link to an open issue here)**
Hard-coded `fig settings [subcommand] [arg]`

**What is the new behavior (if this is a feature change)?**
Use `generateSpec` and the settings schema at `~/.fig/tools/all-settings.json` to dynamically generate the subcommand and args.

**Additional info:**

### CODE REVIEW

File: dev/fig.ts [hunk 1]
```diff
@@ -10,337 +20,44 @@ export const completionSpec: Fig.Spec = {
...
+              default: defaultValue,
+            }) => {
+              const suggestions =
+                type === "boolean"
+                  ? ["true", "false"]
+                  : options?.map((option) => ({
+                      name: option["name"] || option,
+                      description: option["description"] || "",
+                    }));
+              const variadic = type === "multi_select";
```
Comment (mattschrage) [inline dev/fig.ts:45]: When a setting is `multiselect` it must be written as a JSON array. See the existing completion spec, specifically `autocomplete.disableForCommands` for the current approach. 

We could also default to just inserting `'["{cursor}"]'`.

**Review by mattschrage — COMMENTED** [review]
One small fix!

File: dev/fig.ts (hunk 1 above)
Comment (QuiiBz) [reply dev/fig.ts:45]: Added the `insertValue` and generators `devCompletionsFolder` & `disableForCommands`!

File: dev/fig.ts [hunk 2]
```diff
@@ -10,337 +71,47 @@ export const completionSpec: Fig.Spec = {
...
+                  : options?.map((option) => ({
+                      name: option["name"] || option,
+                      description: option["description"] || "",
+                    }));
+              const insertValue =
+                type === "multiselect" ? `${name} '{cursor}'` : undefined;
+              const generators = SETTINGS_GENERATOR[name];
+
+              return {
+                name,
```
Comment (mattschrage) [inline dev/fig.ts:101]: Can we add the same icon we had before? eg. `fig://icon?type=commandkey`.

**Review by mattschrage — APPROVED** [review]
One small change and then it looks good to me! (As long as you've tested it with the current `all-settings.json` file and everything works?)

File: dev/fig.ts (hunk 2 above)
Comment (QuiiBz) [reply dev/fig.ts:101]: Done!

File: dev/fig.ts [hunk 3]
```diff
@@ -1,3 +1,64 @@
...
+
+interface Settings {
+  settingName: string;
+  description: string;
+  type: "boolean" | "text" | "single_select" | "multiselect";
+  options?: Array<string | { name: string; description: string }>;
+  default?: string;
+}
+
+const devCompletionsFolderGenerator: Fig.Generator = {
```
Comment (cstrnt) [inline dev/fig.ts:11]: Is there any reason why we don't use the folder template here?

Comment (QuiiBz) [reply dev/fig.ts:11]: The path should be absolute and not relative, so we use a custom generator for this:
Screenshot 2021-06-23 at 17 44 36

**Pull request merged**